tests/system_tests/test_core/test_resume.py

```diff
@@ -99,3 +100,74 @@ def test_resume_output_log(wandb_backend_spy):
             if f.startswith("logs/output_") and f.endswith(".log")
         ]
         assert len(log_files) == 2
+
+
+def test_resume_config_preserves_image_mask(user, wandb_backend_spy):
+    img_array = np.zeros((100, 100, 3), dtype=np.uint8)
+    mask_array = np.zeros((100, 100), dtype=np.uint8)
+    mask_array[30:70, 30:70] = 1
+    class_labels = {1: "square"}
+
+    with wandb.init(project="config_preservation") as run:
+        run.log(
+            {
+                "test_image": wandb.Image(
+                    img_array,
+                    masks={
+                        "prediction": {
+                            "mask_data": mask_array,
+                            "class_labels": class_labels,
+                        }
+                    },
+                )
+            }
+        )
+
+        run_id = run.id
```

Comment:
You can access `run.id` outside of the `with` block, so you don't need to store it.